mPower2/MotorControl/MCTActiveStepViewController.swift

```diff
@@ -0,0 +1,83 @@
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+open class MCTActiveStepViewController : RSDActiveStepViewController {
+    
+    @IBOutlet weak var restartButton: RSDRoundedButton!
+    
+    /// What unit should the unit label display to the user?
+    @IBInspectable
+    open var unitLabelText : String? = nil {
+        didSet {
+            self.unitLabel?.text = unitLabelText
+        }
+    }
+    
+    /// This class overrides `didSet` to update the `countdownLabel` to the new value formatted as a time interval in seconds.
+    /// The `countdownFormatter` is used to format the string derived from this time interval.
+    override open var countdown: Int {
+        didSet {
+            self.countdownLabel?.text = countdownFormatter.string(from: TimeInterval(countdown))
+        }
+    }
+    
+    /// Formatter for the countdown label.
+    /// Overriden to not display minutes when the minutes place would be 0.
+    override lazy open var countdownFormatter : DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+        formatter.allowedUnits = countdown < 60 ? [.second] : [.minute, .second]
+        formatter.unitsStyle = .positional
+        formatter.zeroFormattingBehavior = [ .pad ]
+        return formatter
+    }()
+    
+    /// Override viewWillAppear to also set the unitLabel text.
+    override open func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.unitLabel?.text = "SECONDS REMAINING" // TODO rkolmos 03/30/2018 localize
+        (self.step as? RSDActiveUIStepObject)?.nextStepIdentifier = nil
+    }
+    
+    @IBAction func restartButtonTapped(_ sender: Any) {
+        skipForward()
+    }
+    
+    /// Override skip forward to skip backward to the walk step.
+    override open func skipForward() {
+        guard let activeStep = self.step as? RSDActiveUIStepObject else { return }
+        activeStep.nextStepIdentifier = "walk"
```

Comment:
Since this is a hack (smile), can you add a TODO comment to replace this implementation with refactoring ResearchStack2 to support linking an `RSDUIAction` to navigation? Thanks. 

In doing this, I wanted to keep us moving forward and test a theory (which allowed us to discover some bugs) without the more roundabout architecture, but that doesn't mean it doesn't have a certain "code smell". While it works, it's not really a pattern we want to encourage since the whole idea behind using protocols (instead of object inheritance) is so the implementation doesn't **have** to inherit from `RSDActiveUIStepObject`.